rebar.config.erl

```diff
@@ -386,126 +386,60 @@ overlay_vars_pkg(pkg) ->
-            syntax_tools,
-            ssl,
-            os_mon,
-            inets,
-            compiler,
-            runtime_tools,
-            redbug,
-            xmerl,
-            {hocon, load},
-            telemetry
-        ],
-    DBApps =
-        [mnesia_rocksdb || is_rocksdb_supported()] ++
-            [
-                mnesia,
-                mria,
-                ekka
-            ],
-    BusinessApps =
-        [
-            emqx,
-            emqx_conf,
-
-            esasl,
-            observer_cli,
-            tools,
-            covertool,
-            % started by emqx_machine
-            system_monitor,
-            emqx_utils,
-            emqx_http_lib,
-            emqx_resource,
-            emqx_connector,
-            emqx_authn,
-            emqx_authz,
-            emqx_auto_subscribe,
-            emqx_gateway,
-            emqx_gateway_stomp,
-            emqx_gateway_mqttsn,
-            emqx_gateway_coap,
-            emqx_gateway_lwm2m,
-            emqx_gateway_exproto,
-            emqx_exhook,
-            emqx_bridge,
-            emqx_bridge_mqtt,
-            emqx_bridge_http,
-            emqx_rule_engine,
-            emqx_modules,
-            emqx_management,
-            emqx_dashboard,
-            emqx_retainer,
-            emqx_prometheus,
-            emqx_psk,
-            emqx_slow_subs,
-            emqx_mongodb,
-            emqx_redis,
-            emqx_mysql,
-            emqx_plugins
-        ] ++
-            [quicer || is_quicer_supported()] ++
-            [bcrypt || provide_bcrypt_release(ReleaseType)] ++
-            %% Started automatically when needed (only needs to be started when the
-            %% port implementation is used)
-            [jq || is_jq_supported()] ++
-            [observer || is_app(observer)] ++
-            relx_apps_per_edition(Edition),
+    {ok, [
+        #{
+            db_apps := DBApps0,
+            system_apps := SystemApps,
+            common_business_apps := CommonBusinessApps0,
+            ee_business_apps := EEBusinessApps,
+            ce_business_apps := CEBusinessApps
+        }
+    ]} = file:consult("apps/emqx_machine/priv/reboot_lists.eterm"),
+    DBApps = filter(DBApps0, #{mnesia_rocksdb => is_rocksdb_supported()}),
+    CommonBusinessApps =
+        filter(CommonBusinessApps0, #{
+            quicer => is_quicer_supported(),
+            bcrypt => provide_bcrypt_release(ReleaseType),
+            jq => is_jq_supported(),
+            observer => is_app(observer)
+        }),
+    EditionSpecificApps =
+        case Edition of
+            ee -> EEBusinessApps;
+            ce -> CEBusinessApps
+        end,
+    BusinessApps = CommonBusinessApps ++ EditionSpecificApps,
     SystemApps ++
         %% EMQX starts the DB and the business applications:
         [{App, load} || App <- DBApps] ++
         [emqx_machine] ++
         [{App, load} || App <- BusinessApps].
 
+filter(AppList, Filters) ->
+    lists:foldr(
```

Comment:
Is this foldr equivalent to `lists:filter`?

Reply:
Yes.  I started doing something else, but later it became filter and I forgot to actually use that.